mantidproject/mantid
# Fix binning params in the SofQW* algorithms

The `SofQW*` algorithms failed when they were supplied with 'bin width only' binning parameters. This PR fixes the issue by adding E and Q binning range estimation and by fixing the 'bin width only' case in `VectorHelper::createAxisFromRebinParams`.

**To test:**

The following script should work after these changes are applied. The data can be found in Mantid's unit test data.
```python
DirectILLCollectData(
    Run='ILL/IN6/164192.nxs',
    OutputWorkspace='ws',
)
ws = mtd['ws']
e = ws.run().getProperty('Ei').value
print(e)
ws = ConvertUnits(ws, 'DeltaE', EMode='Direct', EFixed=e)
ws = Rebin(ws, '-10000, 5, 5')

ws = SofQWNormalisedPolygon('ws', QAxisBinning='0.1', EAxisBinning='-10000, 5, 5', EMode='Direct', EFixed=e)
```

Fixes #22140.

**Release Notes** 

The Direct Inelastic release notes were updated.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
